### ISSUE TITLE
Make the Realm class implement an interface so it can be mocked

### ISSUE THREAD

**Issue by kristiandupont** [realm/realm-dotnet#537]

**Comment by AndyDentFree** [comment]
I don't think we need an interface so it can be mocked, we just need a mocking assembly declaring the same objects. If we have other reasons for wanting an interface, fine, but I don't think mocking alone is enough to justify it.

**Comment by AndyDentFree** [comment]
This is effectively a duplicate of #529

**Comment by kristiandupont** [comment]
Reopened as it turned out that it's actually a different thing. This is about adding an interface to the `Realm` class, not `RealmObject`.

**Comment by joseprl89** [comment]
Interested on this

**Comment by willsb** [comment]
Is this on the Roadmap? Can I offer some help to implement this? :)

**Comment by nirinchev** [comment]
We haven't prioritized this, but PR's are always welcome :)
